privacyidea/lib/auditmodules/sqlaudit.py

```diff
@@ -223,6 +223,11 @@ def finalize_log(self):
             self.audit_data["policies"] = ",".join(self.audit_data.get("policies",[]))
             if self.config.get("PI_AUDIT_SQL_TRUNCATE"):
                 self._truncate_data()
+            if "tokentype" in self.audit_data:   # pragma: no cover
+                log.warning("We have a wrong 'tokentype' key. This should not happen. Fix it!. "
+                            "Error occurs in action: {0!r}.".format(self.audit_data.get("action")))
+                if not "token_type" in self.audit_data:
+                    self.audit_data["token_type"] = self.audit_data.get("tokentype")
```

Comment:
I think this approach is fine, but I think it would be nice if you could add a small test for the branch that rewrites ``tokentype`` -> ``token_type``. It could be as easy as passing a ``tokentype`` parameter and checking that the resulting ``LogEntry`` has the ``token_type`` field set correctly. What do you think?

Reply:
Will do.

Reply:
done in 3442b31f

Reply:
thanks! seems like the tests are failing now because the commit happens to contain ``audit.sqlite``. I'll fix this.

Reply:
darn. I always try to remove these changing testdata/ Thanks!